### ISSUE TITLE
Missing data 

### ISSUE THREAD

**Issue by wildintellect** [MAAP-Project/gedi-subsetter#40]
Reported by @abarenblitt The following run of GEDI Subsetter isn't returning as much data as expected. Results are the same if limit is 10 or 250. Results are returned for 4 of the 5 polygons.
```
import os
from maap.maap import MAAP
maap = MAAP(maap_host='api.ops.maap-project.org')

inputs = dict(
   aoi="https://maap-ops-workspace.s3.amazonaws.com/shared/abarenblitt/PA_Subset_Senegal.geojson",
   doi="10.5067/GEDI/GEDI02_B.002",
   lat="geolocation/lat_lowestmode",
   lon="geolocation/lon_lowestmode",
   beams="all",
   columns="rh100",
   query="l2a_quality_flag == 1 and l2b_quality_flag == 1 and sensitivity > 0.95",
   limit = 250
)

result = maap.submitJob(
    identifier="gedi-subset",
    algo_id="gedi-subset_ubuntu",
    version="0.5.0",
    queue="maap-dps-worker-32gb",
    username="abarenblitt",
    **inputs
)
```
Google Earth Engine Results
![image](https://github.com/MAAP-Project/gedi-subsetter/assets/306179/0e9cd2c0-8abd-4574-80e8-3fc22a63bc9d)

MAAP Results
![image](https://github.com/MAAP-Project/gedi-subsetter/assets/306179/f6ef3461-d542-467b-b213-063b417bad73)

AOI GeoJson and the Results GPKG from subsetter that corresponds with the image above.
[PA_Subset_Senegal.zip](https://github.com/MAAP-Project/gedi-subsetter/files/11491800/PA_Subset_Senegal.zip)

## Ideas
- Could it be the query filters?
- Is the multiple polygons in the GeoJson and issue?
- Verify the number of granules that should be used. Currently 13 granules are used even if the limit is 10?

**Comment by chuckwondo** [comment]
@wildintellect, are those images correctly labeled? The one labeled MAAP is much more complete than the one labeled GEE, but the issue seems to indicate that the situation is the other way around, no?

**Comment by wildintellect** [comment]
Current Granules in these results:
```
array(['GEDI02_B_2022326152906_O22344_02_T10432_02_003_01_V002.h5',
       'GEDI02_B_2022337204245_O22518_03_T09413_02_003_01_V002.h5',
       'GEDI02_B_2021116031237_O13421_02_T10432_02_003_01_V002.h5',
       'GEDI02_B_2021235040817_O15266_02_T04740_02_003_01_V002.h5',
       'GEDI02_B_2021330235457_O16751_03_T10989_02_003_01_V002.h5',
       'GEDI02_B_2022182004445_O20101_02_T07586_02_003_02_V002.h5',
       'GEDI02_B_2019288081641_O04758_02_T01894_02_003_01_V002.h5',
       'GEDI02_B_2020065094327_O06961_03_T05297_02_003_01_V002.h5',
       'GEDI02_B_2020073063548_O07083_03_T02451_02_003_01_V002.h5',
       'GEDI02_B_2020085015548_O07266_03_T03721_02_003_01_V002.h5',
       'GEDI02_B_2020320045530_O10911_03_T07990_02_003_01_V002.h5',
       'GEDI02_B_2020324032221_O10972_03_T06567_02_003_01_V002.h5',
       'GEDI02_B_2020347180202_O11338_03_T09413_02_003_01_V002.h5'],
      dtype=object)
```

**Comment by chuckwondo** [comment]
This turns out to be due to the GeoJSON containing multiple polygons. This is not due to a singe MultiPolygon, but rather multiple separate Polygons.

The bug is in the part of the code that filters the results obtained from CMR, which uses only the _first_ geometry in the GeoJSON for constructing a bounding box for the CMR query.

**Comment by chuckwondo** [comment]
Fixed in v0.6.0, #39
